components/Collections/examples/CollectionsCellSeparatorExample.m

```diff
@@ -80,4 +85,44 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView
   return cell;
 }
 
+- (UICollectionReusableView *)collectionView:(UICollectionView *)collectionView
+           viewForSupplementaryElementOfKind:(NSString *)kind
+                                 atIndexPath:(NSIndexPath *)indexPath {
+  MDCCollectionViewTextCell *supplementaryView =
+      [collectionView dequeueReusableSupplementaryViewOfKind:kind
+                                         withReuseIdentifier:kind
+                                                forIndexPath:indexPath];
+
+  if ([kind isEqualToString:UICollectionElementKindSectionHeader]) {
+    if (indexPath.section == 3) {
+      supplementaryView.textLabel.text = @"Header with separator";
+    } else if (indexPath.section == 4) {
+      supplementaryView.textLabel.text = @"Header without separator";
+    }
+  }
+
+  return supplementaryView;
+}
+
+- (CGSize)collectionView:(UICollectionView *)collectionView
+                             layout:(UICollectionViewLayout *)collectionViewLayout
+    referenceSizeForHeaderInSection:(NSInteger)section {
+  if (section == 3 || section == 4) {
+    return CGSizeMake(collectionView.bounds.size.width, MDCCellDefaultOneLineHeight);
+  }
+  return CGSizeZero;
+}
+
+#pragma mark - <MDCCollectionViewStylingDelegate>
+
+- (BOOL)collectionView:(UICollectionView *)collectionView
+    shouldHideItemSeparatorAtIndexPath:(NSIndexPath *)indexPath {
```

Comment:
I am already using a 4 space indentation, and this is what clang-format is giving. It also seems consistent with the rest of the methods.